app/lib/multistep/form.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+require_dependency "multistep/builder"
+require_dependency "multistep/step"
+
+module Multistep
+  module Form
+    extend ActiveSupport::Concern
+
+    included do
+      prepend ValidationContext
+      include ActiveModel::Attributes
+      include ActiveModel::Serialization
+
+      attribute :completed_steps, default: []
+    end
+
+    class_methods do
+      def step(step_name, &block)
+        builder = Builder.new(step_name: step_name, form_class: self)
+        builder.instance_exec(&block) if block
+
+        steps[step_name] = builder.to_step
+      end
+
+      def steps
+        @steps ||= {}
+      end
+
+      def step_route_constraint
```

Comment:
Is `form` the place that this should live?

Reply:
Yup, probably not! I think I'd prefer another module `Multistep::Route` that is responsible for drawing all the routes with a single call, making this method its private implementation detail. I'll get this added.